lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -116,6 +124,190 @@ struct ConvertRLWEDecrypt : public OpConversionPattern<RLWEDecryptOp> {
+                  "Unsupported encoding attribute for cleartext start");
+              return failure();
+            });
+
+    // Should return failure if cleartextBitwidth or cleartextStart fail.
+    if (failed(cleartextBitwidthOrFailure) || failed(cleartextStartOrFailure)) {
+      return failure();
+    }
+    auto cleartextBitwidth = cleartextBitwidthOrFailure.value();
+    auto cleartextStart = cleartextStartOrFailure.value();
+
+    // Check that cleartext_start = cleartext_bitwidth for BGV encryption.
+    if (cleartextBitwidth != cleartextStart) {
+      // TODO (#882): Add support for other encryption schemes besides BGV. Left
+      // as future work.
+      op.emitError() << "`lwe.rlwe_encrypt` expects BGV encryption"
+                     << " with cleartext_start = cleartext_bitwidth, but"
+                     << " found cleartext_start = " << cleartextStart
+                     << " and cleartext_bitwidth = " << cleartextBitwidth
+                     << ".";
+      return failure();
+    }
+
+    auto isPublicKey =
+        llvm::TypeSwitch<Type, bool>(op.getKey().getType())
+            .Case<lwe::RLWEPublicKeyType>([](auto key) -> bool { return true; })
+            .Case<lwe::RLWESecretKeyType>(
+                [](auto key) -> bool { return false; })
+            .Default([](Type key) -> bool {
+              llvm_unreachable(
+                  "Unsupported key type: Neither public key nor secret key");
+              return false;
+            });
+
+    ImplicitLocOpBuilder builder(loc, rewriter);
+    if (isPublicKey) {
+      auto index0 = builder.create<arith::ConstantIndexOp>(0);
+      tensor::ExtractOp publicKey0 =
+          builder.create<tensor::ExtractOp>(key, ValueRange{index0});
+      auto index1 = builder.create<arith::ConstantIndexOp>(1);
+      tensor::ExtractOp publicKey1 =
+          builder.create<tensor::ExtractOp>(key, ValueRange{index1});
+
+      auto dimension =
+          inputT.getRing().getPolynomialModulus().getPolynomial().getDegree();
+
+      auto elementType = rewriter.getIntegerType(inputT.getRing()
+                                                     .getCoefficientModulus()
+                                                     .getType()
+                                                     .getIntOrFloatBitWidth());
+
+      auto tensorParams = RankedTensorType::get({dimension}, elementType);
+
+      // constantT is 2**(cleartextBitwidth), and is used for scalar
+      // multiplication.
+      // TODO(#876): Migrate to using the plaintext modulus of the encoding info
+      // attributes.
+      auto constantT = builder.create<arith::ConstantOp>(
+          builder.getI32IntegerAttr(1 << cleartextBitwidth));
+
+      // TODO (#881): Add pass options to change the seed (which is currently
+      // hardcoded to 0).
+
+      // Initialize random number generator with seed.
+      auto generateRandom =
+          builder.create<random::InitOp>(index0, builder.getI32IntegerAttr(32));
```

Comment:
hardcode this to 0 per the comment above?
```suggestion
          builder.create<random::InitOp>(index0, builder.getI32IntegerAttr(0));
```

Reply:
No, index0 is the seed. The builder.getI32IntegerAttr(32) is actually the number of bits attribute which might end up being removed. I'll add a TODO to update this as Random Dialect get cleaned up.

Reply:
Oh, thank you! Got it.